parquet-column/src/main/java/org/apache/parquet/column/statistics/Statistics.java

```diff
@@ -139,6 +140,43 @@ public Statistics<?> build() {
     }
   }
 
+  // Builder for FLOAT16 type to handle special cases of min/max values like NaN, -0.0, and 0.0
+  private static class Float16Builder extends Builder {
+    public Float16Builder(PrimitiveType type) {
+      super(type);
+      assert type.getPrimitiveTypeName() == PrimitiveTypeName.BINARY;
+    }
+
+    @Override
+    public Statistics<?> build() {
+      Float16Statistics stats = (Float16Statistics) super.build();
+      if (stats.hasNonNullValue()) {
+        Binary bMin = stats.genericGetMin();
+        Binary bMax = stats.genericGetMax();
+        short min = Float16.fromBytesLittleEndian(bMin.getBytes());
+        short max = Float16.fromBytesLittleEndian(bMax.getBytes());
+        // Drop min/max values in case of NaN as the sorting order of values is undefined for this case
+        if (Float16.isNaN(min) || Float16.isNaN(max)) {
+          bMin = Binary.fromConstantByteArray(Float16.toBytesLittleEndian(Float16.POSITIVE_ZERO));
```

Comment:
It seems worth adding two static constants (of Binary type) to Float16 for POSITIVE_ZERO and NEGATIVE_ZERO as they are repeatedly constructed.